src/main/java/spoon/template/Substitution.java

```diff
@@ -663,9 +663,27 @@ public static void redirectTypeReferences(CtElement element, CtTypeReference<?>
 		if (c.isShadow()) {
 			throw new SpoonException("The template " + template.getClass().getName() + " is not part of model. Add template sources to spoon template path.");
 		}
+		checkTemplateContracts(c);
 		return c;
 	}
 
+	private static <T> void checkTemplateContracts(CtClass<T> c) {
+		for (CtField f : c.getFields()) {
+			if (f.getAnnotation(Parameter.class) != null && !f.getAnnotation(Parameter.class).value().equals("")) {
```

Comment:
Please don't use f.getAnnotation(Class) several times in a row.
This API uses reflection and proxy, the performances are thus not good at all.